### PR TITLE
[mypyc] support unicode identifiers

### PR DESCRIPTION
**Pull request by r3m0t**
### Description

Fixes mypyc/mypyc#699

### CODE REVIEW

File: mypyc/namegen.py [hunk 1]
```diff
@@ -30,6 +40,9 @@ class NameGenerator:
       collides '___' with '.3_', but this is OK because names
       may not start with a digit.)
 
+    * Replace non-ASCII characters with their codepoints, such as '_e9'
+      for U+00E9.
```
Comment (msullivan) [inline mypyc/namegen.py:44]: The downside here is that we'll get name collisions if `_e9` appears in a name. Can we come up with an encoding scheme that dodges this without uglifying normal names too much, like we did for `.` -> `___`. Would just using `___3[the unicode code point]_` work?

File: mypyc/namegen.py [hunk 2]
```diff
@@ -36,9 +40,7 @@ class NameGenerator:
       prefixes will be 'bar_foo_' and 'baz_foo_'.
 
     * Replace '.' in the Python name with '___' in the C name. (And
-      replace the unlikely but possible '___' with '___3_'. This
-      collides '___' with '.3_', but this is OK because names
-      may not start with a digit.)
+      replace '___' with '____' and so on to prevent collisions)
```
Comment (msullivan) [inline mypyc/namegen.py:43]: This doesn't seem to help with the unicode collisions, though?

Comment (msullivan) [reply mypyc/namegen.py:43]: Oh, and it introduces new collisions. `a._` and `a___` (3 underscores) now both map to `a____` (4 underscores), for example.

Comment (r3m0t) [reply mypyc/namegen.py:43]: Oops!

**Review by msullivan — CHANGES_REQUESTED** [review]
I think restoring the old `___` -> `___3_` encoding and then mapping unicode to `___3[unicode hex]_` would work. There would be collisions with `.3[stuff]_` but that's OK because names can't start with digits.

**Pull request closed without merging**